x-pack/platform/plugins/private/reporting/public/management/integrations/scheduled_report_share_integration.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import type { ShareContext } from '@kbn/share-plugin/public';
+import type { ExportShareDerivatives } from '@kbn/share-plugin/public/types';
+import type { ReportingSharingData } from '@kbn/reporting-public/share/share_context_menu';
+import { EuiButton } from '@elastic/eui';
+import type { ReportingAPIClient } from '@kbn/reporting-public';
+import { HttpSetup } from '@kbn/core-http-browser';
+import { getKey as getReportingHealthQueryKey } from '../hooks/use_get_reporting_health_query';
+import { queryClient } from '../../query_client';
+import { ScheduledReportFlyoutShareWrapper } from '../components/scheduled_report_flyout_share_wrapper';
+import { SCHEDULE_EXPORT_BUTTON_LABEL } from '../translations';
+import type { ReportingPublicPluginSetupDependencies } from '../../plugin';
+import { getReportingHealth } from '../apis/get_reporting_health';
+
+export interface CreateScheduledReportProviderOptions {
+  apiClient: ReportingAPIClient;
+  services: ReportingPublicPluginSetupDependencies;
+}
+
+export const shouldRegisterScheduledReportShareIntegration = async (http: HttpSetup) => {
+  const { isSufficientlySecure, hasPermanentEncryptionKey } = await queryClient.fetchQuery({
+    queryKey: getReportingHealthQueryKey(),
+    queryFn: () => getReportingHealth({ http }),
+  });
+  return isSufficientlySecure && hasPermanentEncryptionKey;
+};
+
+export const createScheduledReportShareIntegration = ({
+  apiClient,
+  services,
+}: CreateScheduledReportProviderOptions): ExportShareDerivatives => {
+  return {
+    id: 'scheduledReports',
+    groupId: 'exportDerivatives',
+    shareType: 'integration',
+    config: (shareOpts: ShareContext): ReturnType<ExportShareDerivatives['config']> => {
+      const { sharingData } = shareOpts as unknown as { sharingData: ReportingSharingData };
+      return {
+        label: ({ openFlyout }) => (
+          <EuiButton iconType="calendar" onClick={openFlyout}>
+            {SCHEDULE_EXPORT_BUTTON_LABEL}
+          </EuiButton>
+        ),
+        flyoutContent: ({ closeFlyout }) => {
+          return (
+            <ScheduledReportFlyoutShareWrapper
+              apiClient={apiClient}
+              services={services}
+              sharingData={sharingData}
+              onClose={closeFlyout}
+            />
+          );
+        },
+        flyoutSizing: { size: 'm', maxWidth: 500 },
+      };
+    },
+    prerequisiteCheck: ({ license }) => {
+      if (!license) {
+        return false;
+      }
+      return license.type !== 'basic';
```

Comment:
We added an allowlist for licenses here: https://github.com/elastic/kibana/pull/221028/commits/f5f9d9daedcd18447b6a02335a53631a44413788 Not sure if you can reuse.

Reply:
I couldn't import it from there since it's marked as server code so I moved it to the `common` folder, hope it's fine 🙂